### PR TITLE
remove secretKey from extractor log

### PR DESCRIPTION
**Pull request by bingzhang**
## Review Time Estimate

- [ ] Immediately
- [ ] Within one week
- [x] When possible

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the CHANGELOG.md.
- [ ] I have signed the CLA
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: app/services/GraylogService.scala [hunk 1]
```diff
@@ -49,7 +49,18 @@ class GraylogService @Inject() extends LogService {
           val messagesjsonValue = responseJsonValue.get("messages")
           messagesjsonValue.foreach(elem => {
             val elemMap = elem.asInstanceOf[java.util.HashMap[String, java.util.HashMap[String, String]]]
-            val message: String = elemMap.get("message").get("message")
+            var message: String = elemMap.get("message").get("message")
```
Comment (robkooper) [inline app/services/GraylogService.scala:52]: what if message does not exist in the map
can we just use regex to search/replace

**Review by robkooper — CHANGES_REQUESTED** [review]

File: app/services/GraylogService.scala (hunk 1 above)
Comment (bingzhang) [reply app/services/GraylogService.scala:52]: if the messages is not empty. message is not empty. If we worry about the exception, we do try catch around it.

File: app/services/GraylogService.scala [hunk 2]
```diff
@@ -48,17 +48,32 @@ class GraylogService @Inject() extends LogService {
...
+                case -1 => {
+                  logs = logs :+ message
+                }
+                case _ => {
+                  val filterMessage: String = message.substring(0, keyIndex)
+                  logs = logs :+ filterMessage
+                }
               }
+            } catch {
+              case e: Exception => Logger.warn(s"Exception: '$message'")
```
Comment (bodom0015) [inline app/services/GraylogService.scala:76]: I'm getting a compilation error here: `not found: value message`

It looks like `var message` is defined within the `try` block (line 53), which goes out of scope before the `catch` block executes. Perhaps you meant `Logger.warn(s"Exception: '$e.getMessage()'")`? See https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#getMessage--

Comment (robkooper) [reply app/services/GraylogService.scala:76]: github actions does the same, please fix @bingzhang

**Comment by robkooper** [comment]
This code will need some cleanup, but that can be done in the future.

Right now I think it does what it needs to do but can use some cleanup to leverage of more scala options, such as `message.foreach()` option etc. Also using a regex might simplify the code instead of us trying to implement a regex (almost).

Finally the code cuts off everything after `key=` in the message, which could be significant more than what we want to delete.

File: app/services/GraylogService.scala (hunk 2 above)
Comment (bingzhang) [reply app/services/GraylogService.scala:77]: there is no build error in my local. I will check the auction.

Comment (bingzhang) [reply app/services/GraylogService.scala:77]: I rerun the action, but it still gives the error.
@bodom0015 can you pull the latest pr, I run `sbt dist` in my local. it has no error.

```
[warn] warning: [options] bootstrap class path not set in conjunction with -source 1.7
[warn] 1 warning
[info] Packaging /Users/bing/github/clowder/target/scala-2.10/clowder_2.10-1.12.0-develop.jar ...
[info] Done packaging.
[info] 
[info] Your package is ready in /Users/bing/github/clowder/target/universal/clowder-1.12.0-develop.zip
[info] 
[success] Total time: 165 s, completed Oct 22, 2020 9:13:12 AM
```

Comment (lmarini) [reply app/services/GraylogService.scala:77]: @bingzhang I cannot run it either. Try running it and pulling up the frontend. Make sure you have pushed all your commits.

Comment (bingzhang) [reply app/services/GraylogService.scala:77]: @lmarini I can run the Clowder GUI from my local.

```
[warn] play - No application found at invoker init
2020-10-22 09:24:15,100 - [INFO   ] - application - Connecting to : mongodb://127.0.0.1:27017/clowder
2020-10-22 09:24:15,281 - [WARN   ] - application - Missing mongo update add-trash-dataset. Application might be broken.
2020-10-22 09:24:15,283 - [WARN   ] - application - Missing mongo update add-trash-collection. Application might be broken.
2020-10-22 09:24:15,285 - [WARN   ] - application - Missing mongo update fixing-typehint-users. Application might be broken.
2020-10-22 09:24:15,287 - [WARN   ] - application - Missing mongo update convert-to-spaces. Application might be broken.
2020-10-22 09:24:15,288 - [WARN   ] - application - Missing mongo update fixing-taglength. Application might be broken.
```

Comment (lmarini) [reply app/services/GraylogService.scala:77]: > @lmarini I can run the Clowder GUI from my local.
> 
> ```
> [warn] play - No application found at invoker init
> 2020-10-22 09:24:15,100 - [INFO   ] - application - Connecting to : mongodb://127.0.0.1:27017/clowder
> 2020-10-22 09:24:15,281 - [WARN   ] - application - Missing mongo update add-trash-dataset. Application might be broken.
> 2020-10-22 09:24:15,283 - [WARN   ] - application - Missing mongo update add-trash-collection. Application might be broken.
> 2020-10-22 09:24:15,285 - [WARN   ] - application - Missing mongo update fixing-typehint-users. Application might be broken.
> 2020-10-22 09:24:15,287 - [WARN   ] - application - Missing mongo update convert-to-spaces. Application might be broken.
> 2020-10-22 09:24:15,288 - [WARN   ] - application - Missing mongo update fixing-taglength. Application might be broken.
> ```

The latest commit fixes the issue.

Comment (bingzhang) [reply app/services/GraylogService.scala:77]:
```
MacBook-Pro:clowder bing$ git branch
  add_graylog_settings
  add_log_settings
  branch-extractor-catalog-wireframe-viewlogs
  branch-extractor-catalog-wireframe-viewmetrics
  clean_extractors_tmpfiles
  develop
  extractor-catalog-wireframe
  fix_mismatch_extractorname
  fix_mismatch_extractorname_release_1.12.0
  improve_simple_test
  release/1.12.0
  remove_debugcode
  remove_debugcode_viewlog
* remove_secretkey
MacBook-Pro:clowder bing$ 
MacBook-Pro:clowder bing$ 
MacBook-Pro:clowder bing$ git status
On branch remove_secretkey
Untracked files:
  (use "git add <file>..." to include in what will be committed)
	scripts/tester/memoryfile
	scripts/tester/test.sh

nothing added to commit but untracked files present (use "git add" to track)
MacBook-Pro:clowder bing$ git diff
MacBook-Pro:clowder bing$ 
```

Comment (bingzhang) [reply app/services/GraylogService.scala:77]: I already committed the code.

Comment (bingzhang) [reply app/services/GraylogService.scala:77]: > > @lmarini I can run the Clowder GUI from my local.
> > ```
> > [warn] play - No application found at invoker init
> > 2020-10-22 09:24:15,100 - [INFO   ] - application - Connecting to : mongodb://127.0.0.1:27017/clowder
> > 2020-10-22 09:24:15,281 - [WARN   ] - application - Missing mongo update add-trash-dataset. Application might be broken.
> > 2020-10-22 09:24:15,283 - [WARN   ] - application - Missing mongo update add-trash-collection. Application might be broken.
> > 2020-10-22 09:24:15,285 - [WARN   ] - application - Missing mongo update fixing-typehint-users. Application might be broken.
> > 2020-10-22 09:24:15,287 - [WARN   ] - application - Missing mongo update convert-to-spaces. Application might be broken.
> > 2020-10-22 09:24:15,288 - [WARN   ] - application - Missing mongo update fixing-taglength. Application might be broken.
> > ```
> 
> The latest commit fixes the issue.

thanks for the check. that is what I mean. I rerun the action on this latest commit, but it is still failed.

**Comment by bingzhang** [comment]
@robkooper is there any fix I will do before we merge this pr?

**Pull request merged**